client/app/services/utils.js

```diff
@@ -0,0 +1,14 @@
+function absoluteUrl(url) {
+  const location = window.location;
+
+  const scheme = location.protocol.toLowerCase();
+  const host = location.host;
+
+  return scheme + '//' + host + url;
+}
+
+export default function init(ngModule) {
+  ngModule.factory('Utils', () => ({
```

Comment:
We don't need Angular's DI here. We can just `export` `absoluteUrl` and then where we need to use it:

```js
import { absoluteUrl } from '@/services/utils';
```